### DISCUSSION TITLE
How do I get commands.has_permissions to return True or False

### DISCUSSION THREAD

**Discussion by prsh11** [Rapptz/discord.py#7131] [Q&A] [answered]
I am trying to get comands.has_permissions to return True or false but it keeps returning decorator object

**Comment by Kev-in123** [comment, marked as answer]
comands.has_permissions is used as a decorator  to check if the user who ran the command has the permissions to run the command
example:
```
@commands.command(description="kick someone")
@commands.guild_only()
@commands.has_permissions(kick_members=True)
async def kick(self, ctx, member : discord.Member,*,reason=None):
    await member.kick(reason=reason)
```

**Comment by PythonCoderAS** [comment]
You can trigger it yourself manually by doing `coommands.guild_only().predicate(ctx)`, where `ctx` is an instance of the Context class.

**Reply by prsh11** [reply]
> You can trigger it yourself manually by doing `coommands.guild_only().predicate(ctx)`, where `ctx` is an instance of the Context class.

I want to check if the perms but manually. This does not seem to work

**Reply by wasi-master** [reply]
> I want to check if the perms but manually. This does not seem to work

Then your question should have been that. This is a xy problem where someone wants to do x but they don’t know how to do x and they think doing y would fix that, and they ask for help with y instead. You should always ask the question and say what you are trying to achieve

**Comment by Phxntxm** [comment]
These aren't meant to be called directly, there's no reason to. Just use the interfaces that exist for this. [ctx.channel.permissions_for](https://discordpy.readthedocs.io/en/stable/api.html#discord.TextChannel.permissions_for) and [ctx.author.guild_permissions](https://discordpy.readthedocs.io/en/stable/api.html#discord.Member.guild_permissions)
